scan/lib/scan/report_collector.rb

```diff
@@ -14,16 +14,22 @@ def initialize(open_report, output_types, output_directory, use_clang_report_nam
     def parse_raw_file(path)
       UI.user_error!("Couldn't find file at path '#{path}'") unless File.exist?(path)
 
+      success = true
       commands = generate_commands(path)
       commands.each do |output_path, command|
-        system(command)
-        UI.success("Successfully generated report at '#{output_path}'")
+        if system(command)
+          UI.success("Successfully generated report at '#{output_path}'")
+        else
+          UI.error("Failed to generate report at '#{output_path}'")
+          success = false
```

Comment:
Yes!

The only advantage of this method is that it reports all the errors for different output types. But we can stop at the first report generation error, this will be less code.